core/src/main/java/nu/yona/server/analysis/service/WeekActivityDTO.java

```diff
@@ -84,4 +98,63 @@ static WeekActivityDTO createInstance(WeekActivity weekActivity, LevelOfDetail l
 								dayActivity -> DayActivityDTO.createInstance(dayActivity, levelOfDetail))),
 				weekActivity.hasPrevious(), weekActivity.hasNext());
 	}
+
+	public static WeekActivityDTO createInstanceInactivity(UserAnonymizedDTO userAnonymized, Goal goal, ZonedDateTime startOfWeek,
```

Comment:
That's not a bad idea, but I prefer to postpone this now and take that along when further refactoring ``ActivityService`` (I'm still unhappy with that class).